### ISSUE TITLE
GHC 9.6 support

### ISSUE THREAD

**Issue by penelopeysm** [discord-haskell/discord-haskell#175]
Hello!

I'm wondering what it would take for GHC 9.6 support (`base == 4.18.0.0`) to be added. The following are the minimal changes needed to get the `discord-haskell` package to build successfully; however, I haven't tested it in any way (e.g. by running the executables).

Do you have a standard procedure for bumping dependencies and testing? Or is this enough? I'm happy to look into this further and make a PR (unless it gets too involved).

```diff
diff --git a/discord-haskell.cabal b/discord-haskell.cabal
index 2e7dcec..71ebe48 100644
--- a/discord-haskell.cabal
+++ b/discord-haskell.cabal
@@ -1,6 +1,6 @@
 cabal-version:       2.0
 name:                discord-haskell
-version:             1.15.5
+version:             1.16.0
 description:         Functions and data types to write discord bots.
                      Official discord docs <https://discord.com/developers/docs/reference>.
                      .
@@ -159,8 +159,8 @@ library
   build-depends:
   -- https://gitlab.haskell.org/ghc/ghc/-/wikis/commentary/libraries/version-history
   -- below also sets the GHC version effectively. set to == 8.10.*, == 9.0.*., == 9.2.*, == 9.4.*
-    base == 4.14.* || == 4.15.* || == 4.16.* || == 4.17.*,
-    aeson >= 1.5 && < 1.6 || >= 2.0 && < 2.2,
+    base == 4.14.* || == 4.15.* || == 4.16.* || == 4.17.* || == 4.18.*,
+    aeson >= 1.5 && < 1.6 || >= 2.0 && < 2.3,
     async >=2.2 && <2.3,
     bytestring >=0.10 && <0.12,
     base64-bytestring >=1.1 && <1.3,
@@ -169,7 +169,7 @@ library
     emoji ==0.1.*,
     http-client >=0.6 && <0.8,
     iso8601-time >=0.1 && <0.2,
-    MonadRandom >=0.5 && <0.6,
+    MonadRandom >=0.5 && <0.7,
     req >=3.9 && <3.14,
     safe-exceptions >=0.1 && <0.2,
     text >=1.2 && <3,
@@ -177,7 +177,7 @@ library
     websockets >=0.12 && <0.13,
     network >=3.0.0.0 && <3.2.0.0,
     wuss >=1.1 && <3,
-    mtl >=2.2 && <2.3,
+    mtl >=2.2 && <2.4,
     unliftio >=0.2 && <0.3,
     scientific >=0.3 && <0.4,
     http-api-data >=0.4 && <0.6,
diff --git a/src/Discord.hs b/src/Discord.hs
index 5ed8bcf..d96fe48 100644
--- a/src/Discord.hs
+++ b/src/Discord.hs
@@ -26,7 +26,8 @@ module Discord
 
 import Prelude hiding (log)
 import Control.Exception (Exception)
-import Control.Monad.Reader (ReaderT, runReaderT, void, ask, liftIO, forever, asks)
+import Control.Monad (void, forever)
+import Control.Monad.Reader (ReaderT, runReaderT, ask, liftIO, asks)
 import Data.Aeson (FromJSON)
 import Data.Default (Default, def)
 import Data.IORef (writeIORef)
```

Apart from the changes above, `cabal outdated` also suggests the following which could be bumped:

```
Outdated dependencies:
bytestring >=0.10 && <0.12 (latest: 0.12.0.0)
http-api-data >=0.4 && <0.6 (latest: 0.6)
time >=1.9 && <1.12 (latest: 1.12.2)
```

**Comment by L0neGamer** [comment]
Ideally we'd also want the matrix build ci thing to include 9.6. If you're willing to make the PR, I'd appreciate you update the GHC versions supported comment, and if you can not up the version, I'll do that for a new release.

**Comment by L0neGamer** [comment]
Resolved by #176
